tests/test_parsing.py

```diff
@@ -141,3 +141,314 @@ def test_body(self):
                                                                                  Edge([2])]
 
         assert self.hoa_body.state2edges == state_edges_dict
+
+
+class TestParsingAut3:
+    """Test parsing for tests/examples/aut3."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set the test up."""
+        parser = HOAParser()
+        cls.hoa_obj = parser(open(str(Path(TEST_ROOT_DIR, "examples", "aut3.hoa"))).read())  # type: HOA
+        cls.hoa_header = cls.hoa_obj.header
+        cls.hoa_body = cls.hoa_obj.body
+
+    def test_hoa_header(self):
+        """Test that the HOA header is correct."""
+        hoa_header = HOAHeader("v1",
+                               Acceptance(And([Atom(0, AtomType.INFINITE), Atom(1, AtomType.INFINITE)])),
+                               nb_states=1,
+                               start_states=[0],
+                               acceptance_name="generalized-Buchi 2",
+                               propositions=('a', 'b'),
```

Comment:
The parameter `propositions` in the constructor `HOAHeader` expects a `Optional[Tuple[str]]`, but a `Tuple[str, str,]` is provided.

To make the annotation compatible with variable-length tuples, please use `Tuple[str, ...]` as described [here](https://stackoverflow.com/a/47534998/7150172)

Reply:
(btw, to be clear, the input is correct, just change the type annotation in the `HOAHeader` constructor.)